amqp/src/main/java/io/github/tcdl/config/amqp/AmqpBrokerConfig.java

```diff
@@ -1,20 +1,34 @@
 package io.github.tcdl.config.amqp;
 
+import java.util.Optional;
+
+import io.github.tcdl.config.ConfigurationUtil;
+import io.github.tcdl.exception.ConfigurationException;
+
 import com.typesafe.config.Config;
 
 import static io.github.tcdl.config.ConfigurationUtil.getString;
 
 public class AmqpBrokerConfig {
-
+    
     private final int port;
     private final String host;
+    private Optional<String> username;
```

Comment:
We might have problem here because:

"The key here is the focus on use as a return type. The class is definitively not intended for use as a property of a Java Bean. Witness to this is that Optional does not implement Serializable, which is generally necessary for widespread use as a property of an object."
From http://blog.joda.org/2014/11/optional-in-java-se-8.html